packages/language-support/src/parserWrapper.ts

```diff
@@ -39,11 +40,12 @@ export interface ParsedStatement {
   // A statement needs to be parsed with the .statements() rule because
   // it's the one that tries to parse until the EOF
   ctx: StatementsOrCommandsContext;
-  diagnostics: SyntaxDiagnostic[];
+  syntaxErrors: SyntaxDiagnostic[];
```

Comment:
why rename this? I think even if "syntaxErrors" could be more desriptive, it's nice to match the type name imo

Reply:
It makes the code more readable in my opinion. Especially when we are using web workers and we need that logic of: no syntax errors, then execute the web worker. It reads better than diagnostics.

Another example is what has changed in `syntaxValidation.ts`:

```ts
    const anySyntacticError =
      statements.filter((statement) => statement.syntaxErrors.length !== 0)
        .length > 0;
```

We know what's happening inside better than if it read diagnostics. 

What we could do is rename this type:
```ts
export type SyntaxDiagnostic = Diagnostic & {
  offsets: { start: number; end: number };
};
```

where `Diagnostic` is coming from the `vscode-types`. Any suggestions?

Reply:
I don't feel very strongly about it & agree syntaxError reads better. Part of me was assuming there could be more than errors from a "Diagnostics" though

Reply:
It's beside the point but for the code example you could use `[some](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array/some)` instead, it's more efficient as it stops iterating as soon as it finds a match 

```ts
const anySyntacticError =
      statements.some((statement) => statement.syntaxErrors.length !== 0)
```